elastic/kibana
# [Fleet] Fix loading fields for transform destination index template

## Description 

Resolve https://github.com/elastic/kibana/issues/175331

We have currently two different approach to build mappings for template and datastreams;

* For datastream, we process all fields field and then put the index template https://github.com/elastic/kibana/blob/main/x-pack/plugins/fleet/server/services/epm/elasticsearch/template/install.ts#L512-L519
* For transform we process the fields file one by one and put the new mappings.

That PR fix that by using processing fields for transform destination index index template the same way we do for datastream index template.


## How to test

I added some unit test, to cover the shallow mapping issue.

You can manually try to add some packages with transform and check the mappings looks correct.
I tried a few package myself (`endpoint`, `ti_opencti` and mappings looks correct)
